mito-ai/mito_ai/completions/prompt_builders/agent_execution_prompt.py

```diff
@@ -23,35 +16,48 @@ def create_agent_execution_prompt(md: AgentExecutionMetadata) -> str:
     ai_optimized_cells_str = '\n'.join([f"{cell}" for cell in md.aiOptimizedCells or []])
     rules_str = get_rules_str(md.additionalContext)
     selected_context_str = get_selected_context_str(md.additionalContext)
+    streamlit_status_str = get_streamlit_app_status_str(md.notebookID, md.notebookPath)
     
-
-    streamlit_status_str = get_streamlit_app_status_str(md.notebookID, md.notebookPath)    
+    sections = []
     
-    context_str = f"""Remember to choose the correct tool to respond with.
-
-{rules_str}
-
-
-{JUPYTER_NOTEBOOK_SECTION_HEADING}
-{ai_optimized_cells_str}
-
-{VARIABLES_SECTION_HEADING}
-{variables_str}
-
-{FILES_SECTION_HEADING}
-{files_str}
-
-{STREAMLIT_APP_STATUS_SECTION_HEADING}
-{streamlit_status_str}
-
-{ACTIVE_CELL_ID_SECTION_HEADING}
-{md.activeCellId}
-
-{selected_context_str}
-
-{cell_update_output_str(md.base64EncodedActiveCellOutput is not None)}"""
-
-    task_str = '' if md.input == '' else f"""Your task: 
-{md.input}"""
+    # Add intro text
+    sections.append(SG.Task("Remember to choose the correct tool to respond with."))
+    
+    # Add rules if present
+    if rules_str:
```

Comment:
I think we want to have empty sections here so the agent knows there is no data instead of it just not being provided. We could even have an empty placeholder message in each section if no data is provided